src/ProxyManager/ProxyGenerator/LazyLoadingGhostGenerator.php

```diff
@@ -82,10 +82,17 @@ public function generate(ReflectionClass $originalClass, ClassGenerator $classGe
         $classGenerator->addPropertyFromGenerator($privateProperties);
         $classGenerator->addPropertyFromGenerator($protectedProperties);
 
+        $propertiesFiltered = Properties::fromReflectionClass($originalClass)
+            ->filter(
+                isset($options['skippedProperties'])
+                    ? $options['skippedProperties']
+                    : []
+            );
+
```

Comment:
Also this code should be changed:

```php

$publicProperties = new PublicPropertiesMap($originalClass);
$privateProperties = new PrivatePropertiesMap($originalClass);
$protectedProperties = new ProtectedPropertiesMap($originalClass);
```

I suggest changing the constructor of those to accept a `Properties` instance